packages/datadog-instrumentations/src/helpers/register.js

```diff
@@ -19,16 +19,27 @@ const hooks = require('./hooks')
 const instrumentations = require('./instrumentations')
 const names = Object.keys(hooks)
 const pathSepExpr = new RegExp(`\\${path.sep}`, 'g')
+
+const convertInstrumentationName = (name) => {
+  return name.replaceAll('@', '').replaceAll('-', '').replaceAll('/', '').replaceAll('_', '').toLowerCase()
+}
+
 const disabledInstrumentations = new Set(
-  DD_TRACE_DISABLED_INSTRUMENTATIONS ? DD_TRACE_DISABLED_INSTRUMENTATIONS.split(',') : []
+  DD_TRACE_DISABLED_INSTRUMENTATIONS
+    ? DD_TRACE_DISABLED_INSTRUMENTATIONS.split(',').map(convertInstrumentationName)
+    : []
 )
+const reenabledInstrumentations = new Set()
 
 // Check for DD_TRACE_<INTEGRATION>_ENABLED environment variables
 for (const [key, value] of Object.entries(process.env)) {
   const match = key.match(/^DD_TRACE_(.+)_ENABLED$/)
   if (match && (value?.toLowerCase() === 'false' || value === '0')) {
-    const integration = match[1].toLowerCase()
+    const integration = convertInstrumentationName(match[1])
     disabledInstrumentations.add(integration)
+  } else if (match && (value?.toLowerCase() === 'true' || value === '1')) {
+    const integration = convertInstrumentationName(match[1])
+    reenabledInstrumentations.add(integration)
   }
```

Comment:
```suggestion
if (match && value) {
  const integration = convertInstrumentationName(match[1])
  if (isFalse(value)) {
    disabledInstrumentations.add(integration)
  } else if (isTrue(value)) {
    reenabledInstrumentations.add(integration)
  }
}
````

In case the env is set, we can be quite confident that the user has it actively enabled or disabled. That is why I pulled it to the top. We also have the helper methods for isTrue and isFalse that we could use here.

Reply:
makes sense!